### PR TITLE
test(ServiceProxy): should preserve Proxy bytecode

### PR DESCRIPTION
**Pull request by re1ro**
- [x] `DepositService`, `GasService`: should preserve Proxy bytecode to get the same address across chains

### CODE REVIEW

**Review by milapsheth — APPROVED** [review]

File: test/AxelarDepositService.js [hunk 1]
```diff
@@ -354,5 +357,30 @@ describe('AxelarDepositService', () => {
...
+            await expect(keccak256(DepositReceiver.bytecode)).to.be.equal(
+              expected,
+            );
+        });
+
+        it('should have the same proxy bytecode', async () => {
+            const proxyBytecode = DepositServiceProxy.bytecode;
+            const proxyBytecodeHash = keccak256(proxyBytecode);
+
+            const expected = {
```
Comment (sammy1991106) [inline test/AxelarDepositService.js:377]: These proxy bytecode doesn't look the same while the name of the test case suggests that they should be?

Comment (milapsheth) [reply test/AxelarDepositService.js:377]: Most of the contracts are compiled differently when targeting a different EVM version. Maybe @re1ro can mention if there's a specific opcode where they differ just to get a better idea. But not really expecting the same one to be generated.

File: test/AxelarGateway.js [hunk 2]
```diff
@@ -728,6 +730,42 @@ describe('AxelarGateway', () => {
...
+                istanbul: '0xc68014e297eb42dbde383254ef3129d59528159e6c51b4f9a38f995be1dd451f',
+                berlin: '0xd3a39792ca8d1ce8e5318135ca29d8a7f0b800837726997b132ebc04f88cf9aa',
+                london: '0x0698929742de660596af20d09d04eb91bfe532ef5e2927858e4c4952034967a5',
+            }[EVM_VERSION]
+
+            await expect(keccak256(tokenDeployerFactory.bytecode)).to.be.equal(
+              expectedDeployer,
+            );
+        });
     });
```
Comment (milapsheth) [inline test/AxelarGateway.js:769]: Can you add the AxelarGatewayProxy and gateway implementation bytecode too? While we don't need it for deriving the same address, we do want to be alerted about any unintentional changes

File: test/AxelarDepositService.js (hunk 1 above)
Comment (deanamiel) [reply test/AxelarDepositService.js:377]: That confused me a bit as well, I think changing the test name would help

**Review by milapsheth — APPROVED** [review]

**Review by deanamiel — APPROVED** [review]

**Review by sammy1991106 — APPROVED** [review]

**Pull request merged**